### PR TITLE
Femtoworld - Adding angular correlation analysis 

### PR DESCRIPTION
**Pull request by zchochul**

### CODE REVIEW

**Comment by victor-gonzalez** [comment]
You have to rebase your changes to the latest tag

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h [hunk 1]
```diff
@@ -61,22 +61,31 @@ class FemtoWorldContainer
...
+  /// \param etaBins mT binning for the histograms
+  /// \param phiBins mT binning for the histograms
+  template <typename T1, typename T2>
+  void init(HistogramRegistry* registry, T1& kstarBins, T1& multBins, T1& kTBins, T1& mTBins, T2& phiBins, T2& etaBins)
   {
     mHistogramRegistry = registry;
     std::string femtoObs;
     if constexpr (mFemtoObs == femtoWorldContainer::Observable::kstar) {
       femtoObs = "#it{k*} (GeV/#it{c})";
+      // dphi =
```
Comment (saganatt) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:73]: remove

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h [hunk 2]
```diff
@@ -133,8 +154,11 @@ class FemtoWorldContainer
   static constexpr int mEventType = eventType;                                        ///< Type of the event (same/mixed, according to femtoWorldContainer::EventType)
   float mMassOne = 0.f;                                                               ///< PDG mass of particle 1
   float mMassTwo = 0.f;                                                               ///< PDG mass of particle 2
+  double PHI_LO;
```
Comment (saganatt) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:157]: could you please keep the convention? mPhiLow, for example?

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h [hunk 3]
```diff
@@ -133,8 +154,11 @@ class FemtoWorldContainer
...
+  double PHI_HI;
+  double PIT = 6.28318530717958623;
```
Comment (saganatt) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:159]: you can make it `static constexpr` as it is a constant. And you can also put this below line 30, like a global constant for all femtoworld-related

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h (hunk 1 above)
Comment (saganatt) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:73]:
```suggestion
```

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h (hunk 2 above)
Comment (saganatt) [reply PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:156]:
```suggestion
  double mPhiLow;
```

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h [hunk 4]
```diff
@@ -133,8 +153,11 @@ class FemtoWorldContainer
   static constexpr int mEventType = eventType;                                        ///< Type of the event (same/mixed, according to femtoWorldContainer::EventType)
   float mMassOne = 0.f;                                                               ///< PDG mass of particle 1
   float mMassTwo = 0.f;                                                               ///< PDG mass of particle 2
+  double PHI_LO;
+  double PHI_HI;
```
Comment (saganatt) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:157]:
```suggestion
  double mPhiHigh;
```

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h [hunk 5]
```diff
@@ -133,8 +153,11 @@ class FemtoWorldContainer
...
+  double PIT = 6.28318530717958623;
```
Comment (saganatt) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:158]:
```suggestion
```

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h (hunk 3 above)
Comment (saganatt) [reply PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:158]:
```suggestion
  static constexpr double PIT = 6.28318530717958623;
```

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h [hunk 6]
```diff
@@ -112,6 +121,16 @@ class FemtoWorldContainer
     const float kT = FemtoWorldMath::getkT(part1, mMassOne, part2, mMassTwo);
     const float mT = FemtoWorldMath::getmT(part1, mMassOne, part2, mMassTwo);
 
+    double delta_eta = part1.eta() - part2.eta();
+
+    double delta_phi = part1.phi() - part2.phi();
+    while (delta_phi < PHI_LO) {
+      delta_phi += PIT;
+    }
+    while (delta_phi > PHI_HI) {
```
Comment (saganatt) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:130]:
```suggestion
    while (delta_phi < mPhiLow) {
      delta_phi += PIT;
    }
    while (delta_phi > mPhiHigh) {
```

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h [hunk 7]
```diff
@@ -75,8 +77,14 @@ class FemtoWorldContainer
     framework::AxisSpec kTAxis = {kTBins, "#it{k}_{T} (GeV/#it{c})"};
     framework::AxisSpec mTAxis = {mTBins, "#it{m}_{T} (GeV/#it{c}^{2})"};
 
+    PHI_LO = (-(int)(phiBins / 4) + 0.5) * 2. * TMath::Pi() / phiBins;
+    PHI_HI = 2 * TMath::Pi() + (-(int)(phiBins / 4) + 0.5) * 2. * TMath::Pi() / phiBins;
+
+    framework::AxisSpec phiAxis = {phiBins, PHI_LO, PHI_HI};
```
Comment (saganatt) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:83]:
```suggestion
    mPhiLow = (-(int)(phiBins / 4) + 0.5) * 2. * TMath::Pi() / phiBins;
    mPhiHigh = 2 * TMath::Pi() + (-(int)(phiBins / 4) + 0.5) * 2. * TMath::Pi() / phiBins;

    framework::AxisSpec phiAxis = {phiBins, mPhiLow, mPhiHigh};
```

**Review by saganatt — APPROVED** [review]

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h [hunk 8]
```diff
@@ -75,8 +77,14 @@ class FemtoWorldContainer
     framework::AxisSpec kTAxis = {kTBins, "#it{k}_{T} (GeV/#it{c})"};
     framework::AxisSpec mTAxis = {mTBins, "#it{m}_{T} (GeV/#it{c}^{2})"};
 
+    mPhiLow = (-(int)(phiBins / 4) + 0.5) * 2. * TMath::Pi() / phiBins;
+    mPhiHigh = 2 * TMath::Pi() + (-(int)(phiBins / 4) + 0.5) * 2. * TMath::Pi() / phiBins;
```
Comment (victor-gonzalez) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:81]: In `O2` `CommonConstants/MathConstants.h` you have these constants defined if you want to use them

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h [hunk 9]
```diff
@@ -112,6 +121,16 @@ class FemtoWorldContainer
...
 
+    double delta_eta = part1.eta() - part2.eta();
+
+    double delta_phi = part1.phi() - part2.phi();
+    while (delta_phi < mPhiLow) {
+      delta_phi += PIT;
+    }
+    while (delta_phi > mPhiHigh) {
+      delta_phi -= PIT;
+    }
```
Comment (victor-gonzalez) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:132]: Same comment as above

File: PWGCF/FemtoWorld/Core/FemtoWorldContainer.h [hunk 10]
```diff
@@ -133,8 +153,11 @@ class FemtoWorldContainer
   static constexpr int mEventType = eventType;                                        ///< Type of the event (same/mixed, according to femtoWorldContainer::EventType)
   float mMassOne = 0.f;                                                               ///< PDG mass of particle 1
   float mMassTwo = 0.f;                                                               ///< PDG mass of particle 2
+  double mPhiLow;
+  double mPhiHigh;
+  static constexpr double PIT = 6.28318530717958623;
```
Comment (victor-gonzalez) [inline PWGCF/FemtoWorld/Core/FemtoWorldContainer.h:158]: Same comment as above

**Review by victor-gonzalez — APPROVED** [review]
I approve it but have a look at my comments in case you want to address them in the next iteration

**Comment by saganatt** [comment]
Thank you @victor-gonzalez ! We will let Zuzanna fix it tomorrow.

**Pull request merged**